pkg/sql/fingerprint_span.go

```diff
@@ -163,14 +177,13 @@ func (p *planner) fingerprintSpanFanout(
 				rv.fingerprint = rv.fingerprint ^ localFingerprint
 				rv.Unlock()
 			}
-			return nil
 		})
 	}
+
+	close(spanChannel)
```

Comment:
you can close this as soon as you finish putting spans in the ch, before launching the goroutines, just to be sure it closes even if you panic while launching for some reason.